### ISSUE TITLE
Update SMI docs

### ISSUE THREAD

**Issue by zachaller** [argoproj/argo-rollouts#2354]
SMI ingress needs to have this annotation applied to the ingress objects our docs do not mention this `nginx.ingress.kubernetes.io/service-upstream: "true"`

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

**Comment by skodamarthi** [comment]
@zachaller Can you please clarify where this needs to be added?  Is this the doc that needs to be updated? https://github.com/argoproj/argo-rollouts/blob/master/docs/getting-started/smi/index.md.

**Comment by zachaller** [comment]
Yes that is the doc and the example ingress object found in the same directory https://github.com/argoproj/argo-rollouts/blob/master/docs/getting-started/smi/ingress.yaml

**Comment by skodamarthi** [comment]
@zachaller Do I need to also update https://github.com/argoproj/argo-rollouts/blob/master/docs/getting-started/smi/ingress.yaml to have this annotation along with https://github.com/argoproj/argo-rollouts/blob/master/docs/getting-started/smi/index.md?
